embedded-graphics/src/image/mod.rs

```diff
@@ -1,16 +1,163 @@
 //! Image object
 
+use crate::coord::Coord;
 use crate::drawable::Dimensions;
+use crate::drawable::Drawable;
+use crate::pixelcolor::PixelColor;
+use crate::transform::Transform;
+use crate::unsignedcoord::{ToSigned, UnsignedCoord};
+use core::marker::PhantomData;
 
 mod image16bpp;
 mod image1bpp;
 mod image8bpp;
 mod image_bmp;
 
-/// Raw image trait
-pub trait Image<'a>: Dimensions {
+/// Raw image type
+pub trait ImageType {}
```

Comment:
Nit: I think this could use a better name, but I'm not sure what to go for. Maybe `ImageExt`? I've seen that used in a couple of other crates but it's not very descriptive. `ImageObject` perhaps?

Reply:
I didn't like the name myself but didn't have a better idea at the time.
Aren't traits like `ImageExt` used extend other traits with additional functionality, like the `ReadBytesExt` trait in the ` byteorder` crate? Which isn't really the case here.
IMHO `ImageObject` is as generic as `ImageType` and doesn't reflect that the trait is used to differentiate between different kinds of image.
How about `ImageFormat` or more verbose `ImageDataFormat`?

Reply:
Ah yes, that seems to be the pattern.

Those names don't sound great either, so let's just go with `ImageType` :). AFAICS it's internal anyway so changing the name later shouldn't be too difficult, right?

Reply:
Currently `ImageType` is public. I wasn't sure if this would be useful for users of e-g.

I would suggest to make it private now and expose it late if necessary.

Reply:
Sounds good to me!

Reply:
Simply changing `ImageType` to be only crate public doesn't seem to work because the `Image` would leak the private type. I'll need to look into it some more in the next few day to find a better solution.